crates/red_knot/src/main.rs

```diff
@@ -34,63 +34,47 @@ struct Args {
     #[command(subcommand)]
     pub(crate) command: Option<Command>,
 
-    #[arg(
-        long,
-        help = "Changes the current working directory.",
-        long_help = "Changes the current working directory before any specified operations. This affects the workspace and configuration discovery.",
-        value_name = "PATH"
-    )]
-    current_directory: Option<SystemPathBuf>,
-
-    #[arg(
-        long,
-        help = "Path to the virtual environment the project uses",
-        long_help = "\
-Path to the virtual environment the project uses. \
-If provided, red-knot will use the `site-packages` directory of this virtual environment \
-to resolve type information for the project's third-party dependencies.",
-        value_name = "PATH"
-    )]
+    /// Run the command within the given project directory.
+    ///
+    /// All `pyproject.toml` files will be discovered by walking up the directory tree from the project root, as will the project's virtual environment (`.venv`).
+    ///
+    /// Other command-line arguments (such as relative paths) will be resolved relative to the current working directory."#,
+    #[arg(long, value_name = "PROJECT")]
+    project: Option<SystemPathBuf>,
+
+    /// Path to the virtual environment the project uses.
+    ///
+    /// If provided, red-knot will use the `site-packages` directory of this virtual environment
+    /// to resolve type information for the project's third-party dependencies.
+    #[arg(long, value_name = "PATH")]
     venv_path: Option<SystemPathBuf>,
 
-    #[arg(
-        long,
-        value_name = "DIRECTORY",
-        help = "Custom directory to use for stdlib typeshed stubs"
-    )]
-    custom_typeshed_dir: Option<SystemPathBuf>,
-
-    #[arg(
-        long,
-        value_name = "PATH",
-        help = "Additional path to use as a module-resolution source (can be passed multiple times)"
-    )]
+    /// Custom directory to use for stdlib typeshed stubs.
+    #[arg(long, value_name = "PATH", alias = "custom-typeshed-dir")]
+    typeshed_path: Option<SystemPathBuf>,
```

Comment:
...`typeshed_dir` rather than `typeshed_path`, since it has to be a directory? But I don't care too deeply, you can disregard this if you diagree :-)

Reply:
I'm not very opinionated on this. I'd prefer using `typeshed_directory` but that's somewhat long. It would also require renaming `extra_search_path` to `extra_search_directory` which I think we should not, because it then no longer maps to `sys.path` 

That's why I'm leaning towards keeping it as is.

Reply:
`--typeshed` seems quite nice to me. I agree it's good to keep it concise

Reply:
Okay, I'll then go with typeshed